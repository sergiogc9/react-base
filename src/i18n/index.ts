import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import config from 'config';
import enLocales from './locales/en.json';
import esLocales from './locales/es.json';

const resources = {
	en: { translation: enLocales },
	es: { translation: esLocales }
};

i18n
	.use(LanguageDetector)
	.use(initReactI18next)
	.init({
		debug: config.isDevelopmentEnvironment(),
		resources,
		fallbackLng: 'en'
	});

export default i18n;
