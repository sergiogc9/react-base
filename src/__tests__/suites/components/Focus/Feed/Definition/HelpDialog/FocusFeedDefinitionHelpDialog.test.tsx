import React from "react";
import { shallow } from "enzyme";

import i18n from '@src/lib/i18n';
import { FocusFeedDefinitionHelpDialog } from "@src/components/Focus/Feed/Definition/HelpDialog/FocusFeedDefinitionHelpDialog";

describe("Definition social", () => {
	let wrapper: any;
	let onCloseMock = jest.fn();

	const getComponent = () => <FocusFeedDefinitionHelpDialog
		onClose={onCloseMock}
		i18n={i18n}
		tReady={true}
		t={jest.fn()}
	/>;

	beforeEach(() => {
		onCloseMock = jest.fn();
		wrapper = shallow(getComponent());
	});

	it("Component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with spanish locales", () => {
		wrapper.setProps({ i18n: { ...i18n, language: "es" } });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with french locales", () => {
		wrapper.setProps({ i18n: { ...i18n, language: "fr" } });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot with italian locales", () => {
		wrapper.setProps({ i18n: { ...i18n, language: "it" } });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Close icon click event", () => {
		expect(onCloseMock).toHaveBeenCalledTimes(0);
		wrapper.find("#focusFeedDefinitionHelpDialogCloseBtn").simulate('click');
		expect(onCloseMock).toHaveBeenCalledTimes(1);
	});
});
