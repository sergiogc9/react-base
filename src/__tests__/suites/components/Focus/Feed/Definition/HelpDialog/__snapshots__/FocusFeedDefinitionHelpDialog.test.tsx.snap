// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition social Component snapshot 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover\\" tabindex=\\"-1\\"><div aria-labelledby=\\"focusFeedDefinitionHelpDialogContainer\\" id=\\"focusFeedDefinitionHelpDialogContainer\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content\\"><span aria-hidden=\\"true\\"></span><div id=\\"focusFeedDefinitionHelpDialogContent\\"><div class=\\"definition-help-header\\"><button id=\\"focusFeedDefinitionHelpDialogCloseBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon icon-cancel md-text--inherit\\"></i></button></div><div class=\\"definition-help-content\\"><div class=\\"definition-help-content-title\\"><i class=\\"md-icon material-icons\\">help_outline</i><div class=\\"definition-help-content-title-texts\\"><div class=\\"big-text\\">Feed definition help tips</div><div class=\\"small-text\\">Use these guidelines to improve your feed results</div></div></div><div class=\\"definition-help-content-tables\\"><div class=\\"definition-help-table-header-responsive\\">Operator:</div><table><thead><tr><th>Operator</th><th>Example</th><th>Result</th></tr></thead><tbody><tr><td data-title=\\"Operator:\\">&nbsp;</td><td data-title=\\"Example:\\">launchmetrics</td><td data-title=\\"Result:\\">Will find results containing the word <span class=\\"bold\\">launchmetrics</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">\\"</span></td><td data-title=\\"Example:\\"><span class=\\"highlight\\">\\"</span>fashion brands<span class=\\"highlight\\">\\"</span></td><td data-title=\\"Result:\\">Will find results with the phrase  <span class=\\"bold\\">fashion brands</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">*</span></td><td data-title=\\"Example:\\">dat<span class=\\"highlight\\">*</span></td><td data-title=\\"Result:\\">Will find results containing words like <span class=\\"bold\\">data</span>, <span class=\\"bold\\">date</span>, <span class=\\"bold\\">dataset</span>, <span class=\\"bold\\">datasheet</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">?</span></td><td data-title=\\"Example:\\">dat<span class=\\"highlight\\">?</span></td><td data-title=\\"Result:\\">Will find results containing words like <span class=\\"bold\\">data</span>, <span class=\\"bold\\">date</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">-</span></td><td data-title=\\"Example:\\">asia <span class=\\"highlight\\">-</span>china</td><td data-title=\\"Result:\\">Will find results containing the word  <span class=\\"bold\\">asia</span> that not mention the word <span class=\\"bold\\">china</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">NOT</span></td><td data-title=\\"Example:\\">asia <span class=\\"highlight\\">NOT</span>china</td><td data-title=\\"Result:\\">Will find results containing the word  <span class=\\"bold\\">asia</span> that not mention the word <span class=\\"bold\\">china</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">AND</span></td><td data-title=\\"Example:\\">paris <span class=\\"highlight\\">AND</span> london</td><td data-title=\\"Result:\\">Will find results that mention <span class=\\"bold\\">paris</span> and <span class=\\"bold\\">london</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">OR</span></td><td data-title=\\"Example:\\">paris <span class=\\"highlight\\">OR</span> london</td><td data-title=\\"Result:\\">Will find results containing the word  <span class=\\"bold\\">paris</span> or the word <span class=\\"bold\\">london</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">()</span></td><td data-title=\\"Example:\\"><span class=\\"highlight\\">(</span>europe AND paris<span class=\\"highlight\\">)</span> OR <span class=\\"highlight\\">(</span>europe AND london<span class=\\"highlight\\">)</span></td><td data-title=\\"Result:\\">Will find results that mention   <span class=\\"bold\\">europe</span> and  <span class=\\"bold\\">paris</span> or results that mention <span class=\\"bold\\">europe</span> and  <span class=\\"bold\\">london</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">~</span></td><td data-title=\\"Example:\\">\\"heathrow airport\\"<span class=\\"highlight\\">~</span>3</td><td data-title=\\"Result:\\">Will find results where the words <span class=\\"bold\\">airport</span> and <span class=\\"bold\\">heathrow</span> appear within 3 words of each other.</td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">sen:</span></td><td data-title=\\"Example:\\"><span class=\\"highlight\\">sen:</span>\\"Fashion & Style\\"</td><td data-title=\\"Result:\\">Will find results containing the phrase <span class=\\"bold\\">Fashion & Style</span></td></tr></tbody></table><div class=\\"definition-help-table-header-responsive\\">Example:</div><table><thead><tr><th>Example</th><th>Result</th></tr></thead><tbody><tr><td data-title=\\"Example:\\"><span class=\\"highlight\\">&quot;</span>fashion brands<span class=\\"highlight\\">&quot;</span> england</td><td data-title=\\"Result:\\">Will find results containing the phrase <span class=\\"bold\\">fashion brands</span> and the word <span class=\\"bold\\">england</span></td></tr><tr><td data-title=\\"Example:\\">europe <span class=\\"highlight\\">-</span>london <span class=\\"highlight\\">&quot;</span>airport gatwick<span class=\\"highlight\\">&quot;~</span>3</td><td data-title=\\"Result:\\">Will find results containing the word <span class=\\"bold\\">europe</span>, not containing the word <span class=\\"bold\\">london</span>, and where the words <span class=\\"bold\\">airport</span> and <span class=\\"bold\\">gatwick</span> appear within 3 words of each other.</td></tr><tr><td data-title=\\"Example:\\"><span class=\\"highlight\\">(&quot;</span>fashion show<span class=\\"highlight\\">&quot; AND (&quot;</span>new york<span class=\\"highlight\\">&quot; OR</span> paris<span class=\\"highlight\\">) AND (NOT</span> classic<span class=\\"highlight\\">))</span></td><td data-title=\\"Result:\\">Will find results containing the phrase <span class=\\"bold\\">fashion show</span> and the phrases  <span class=\\"bold\\">new york</span> or <span class=\\"bold\\">paris</span> and not containing the word <span class=\\"bold\\">classic</span></td></tr><tr><td data-title=\\"Example:\\"><span class=\\"highlight\\">(sen:&quot;</span>Fashion & Style<span class=\\"highlight\\">&quot; AND (sen:</span>@beauty<span class=\\"highlight\\"> OR</span> Beauty<span class=\\"highlight\\">)</span></td><td data-title=\\"Result:\\">Will find results containing the phrase <span class=\\"bold\\">Fashion & Style</span> and the words  <span class=\\"bold\\">@beauty</span> or <span class=\\"bold\\">beauty</span></td></tr></tbody></table></div></div></div></section></div></span>"`;

exports[`Definition social Component snapshot with french locales 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover\\" tabindex=\\"-1\\"><div aria-labelledby=\\"focusFeedDefinitionHelpDialogContainer\\" id=\\"focusFeedDefinitionHelpDialogContainer\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content\\"><span aria-hidden=\\"true\\"></span><div id=\\"focusFeedDefinitionHelpDialogContent\\"><div class=\\"definition-help-header\\"><button id=\\"focusFeedDefinitionHelpDialogCloseBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon icon-cancel md-text--inherit\\"></i></button></div><div class=\\"definition-help-content\\"><div class=\\"definition-help-content-title\\"><i class=\\"md-icon material-icons\\">help_outline</i><div class=\\"definition-help-content-title-texts\\"><div class=\\"big-text\\">Feed definition help tips</div><div class=\\"small-text\\">Use these guidelines to improve your feed results</div></div></div><div class=\\"definition-help-content-tables\\"><div class=\\"definition-help-table-header-responsive\\">Opérateur:</div><table><thead><tr><th>Opérateur</th><th>Exemple</th><th>Résultat</th></tr></thead><tbody><tr><td data-title=\\"Opérateur:\\">&nbsp;</td><td data-title=\\"Exemple:\\">launchmetrics</td><td data-title=\\"Résultat:\\">Pour rechercher résultats qui contiennent le mot <span class=\\"bold\\">launchmetrics</span></td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">\\"</span></td><td data-title=\\"Exemple:\\"><span class=\\"highlight\\">\\"</span>fashion brands<span class=\\"highlight\\">\\"</span></td><td data-title=\\"Résultat:\\">Pour rechercher le groupe de mots <span class=\\"bold\\">fashion brands</span></td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">*</span></td><td data-title=\\"Exemple:\\">paris<span class=\\"highlight\\">*</span></td><td data-title=\\"Résultat:\\">Pour rechercher les mots ayant la même racine comme <span class=\\"bold\\">paris</span>, <span class=\\"bold\\">parisien</span>, <span class=\\"bold\\">parisienne</span></td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">?</span></td><td data-title=\\"Exemple:\\">dat<span class=\\"highlight\\">?</span></td><td data-title=\\"Résultat:\\">Pour rechercher des mots comme <span class=\\"bold\\">data</span>, <span class=\\"bold\\">date</span></td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">-</span></td><td data-title=\\"Exemple:\\">asie <span class=\\"highlight\\">-</span>chine</td><td data-title=\\"Résultat:\\">Pour rechercher le mot \\"asie\\" qui exclue le mot \\"chine\\"</td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">NOT</span></td><td data-title=\\"Exemple:\\">asie <span class=\\"highlight\\">NOT</span> chine</td><td data-title=\\"Résultat:\\">Pour rechercher le mot \\"asie\\" qui exclue le mot \\"chine\\"</td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">AND</span></td><td data-title=\\"Exemple:\\">paris <span class=\\"highlight\\">AND</span> london</td><td data-title=\\"Résultat:\\">Pour rechercher les mots <span class=\\"bold\\">paris</span> et <span class=\\"bold\\">londres</span></td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">OR</span></td><td data-title=\\"Exemple:\\">paris <span class=\\"highlight\\">OR</span> londres</td><td data-title=\\"Résultat:\\">Pour rechercher le mot <span class=\\"bold\\">paris<span class=\\"bold\\"> ou le mot <span class=\\"bold\\">londres</span></td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">()</span></td><td data-title=\\"Exemple:\\"><span class=\\"highlight\\">(</span>europe AND paris<span class=\\"highlight\\">)</span> OR <span class=\\"highlight\\">(</span>europe AND londres<span class=\\"highlight\\">)</span></td><td data-title=\\"Résultat:\\">Pour rechercher les mots <span class=\\"bold\\">europe</span> et <span class=\\"bold\\">paris</span> ou les mots <span class=\\"bold\\">europe</span> et <span class=\\"bold\\">londres</span></td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">~</span></td><td data-title=\\"Exemple:\\">\\"aeroports paris\\"<span class=\\"highlight\\">~</span>3</td><td data-title=\\"Résultat:\\">Pour rechercher les mots \\"aeroports\\" et \\"paris\\" séparés au plus par 3 mots</td></tr><tr><td data-title=\\"Opérateur:\\"><span class=\\"highlight\\">sen:</span></td><td data-title=\\"Exemple:\\"><span class=\\"highlight\\">sen:</span>\\"Fashion & Style\\"</td><td data-title=\\"Résultat:\\">Pour trouver les articles contenant la phrase <span class=\\"bold\\">Fashion & Style</span></td></tr></tbody></table><div class=\\"definition-help-table-header-responsive\\">Exemple:</div><table><thead><tr><th>Exemple</th><th>Résultat</th></tr></thead><tbody><tr><td data-title=\\"Exemple:\\"><span class=\\"highlight\\">&quot;</span>fashion brands<span class=\\"highlight\\">&quot;</span> france</td><td data-title=\\"Résultat:\\">Pour rechercher le groupe de mots <span class=\\"bold\\">fashion brands</span> et le mot <span class=\\"bold\\">france</span></td></tr><tr><td data-title=\\"Exemple:\\">europe <span class=\\"highlight\\">-</span>london <span class=\\"highlight\\">&quot;</span>airport gatwick<span class=\\"highlight\\">&quot;~</span>3</td><td data-title=\\"Résultat:\\">Pour trouver les articles contenant le mot <span class=\\"bold\\">europe</span>, mais pas le mot <span class=\\"bold\\">london</span> et où les mots <span class=\\"bold\\">aeroport</span> et <span class=\\"bold\\">gatwick</span> sont séparés au plus par 3 mots</td></tr><tr><td data-title=\\"Exemple:\\"><span class=\\"highlight\\">(&quot;</span>fashion show<span class=\\"highlight\\">&quot; AND (&quot;</span>new york<span class=\\"highlight\\">&quot; OR</span> paris<span class=\\"highlight\\">) AND (NOT</span> classic<span class=\\"highlight\\">))</span></td><td data-title=\\"Résultat:\\">Pour trouver les articles contenant la phrase <span class=\\"bold\\">fashion show</span> et <span class=\\"bold\\">new york</span> ou <span class=\\"bold\\">paris</span>, mais ne contenant pas le mot <span class=\\"bold\\">classic</span></td></tr><tr><td data-title=\\"Exemple:\\"><span class=\\"highlight\\">(sen:&quot;</span>Fashion & Style<span class=\\"highlight\\">&quot; AND (sen:</span>@beauty<span class=\\"highlight\\"> OR</span> Beauty<span class=\\"highlight\\">)</span></td><td data-title=\\"Résultat:\\">Pour trouver les articles contenant la phrase <span class=\\"bold\\">Fashion & Style</span> et <span class=\\"bold\\">@beauty</span> ou <span class=\\"bold\\">beauty</span></td></tr></tbody></table></div></div></div></section></div></span>"`;

exports[`Definition social Component snapshot with italian locales 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover\\" tabindex=\\"-1\\"><div aria-labelledby=\\"focusFeedDefinitionHelpDialogContainer\\" id=\\"focusFeedDefinitionHelpDialogContainer\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content\\"><span aria-hidden=\\"true\\"></span><div id=\\"focusFeedDefinitionHelpDialogContent\\"><div class=\\"definition-help-header\\"><button id=\\"focusFeedDefinitionHelpDialogCloseBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon icon-cancel md-text--inherit\\"></i></button></div><div class=\\"definition-help-content\\"><div class=\\"definition-help-content-title\\"><i class=\\"md-icon material-icons\\">help_outline</i><div class=\\"definition-help-content-title-texts\\"><div class=\\"big-text\\">Feed definition help tips</div><div class=\\"small-text\\">Use these guidelines to improve your feed results</div></div></div><div class=\\"definition-help-content-tables\\"><div class=\\"definition-help-table-header-responsive\\">Operator:</div><table><thead><tr><th>Operator</th><th>Example</th><th>Result</th></tr></thead><tbody><tr><td data-title=\\"Operator:\\">&nbsp;</td><td data-title=\\"Example:\\">launchmetrics</td><td data-title=\\"Result:\\">Will find results containing the word <span class=\\"bold\\">launchmetrics</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">\\"</span></td><td data-title=\\"Example:\\"><span class=\\"highlight\\">\\"</span>fashion brands<span class=\\"highlight\\">\\"</span></td><td data-title=\\"Result:\\">Will find results with the phrase  <span class=\\"bold\\">fashion brands</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">*</span></td><td data-title=\\"Example:\\">dat<span class=\\"highlight\\">*</span></td><td data-title=\\"Result:\\">Will find results containing words like <span class=\\"bold\\">data</span>, <span class=\\"bold\\">date</span>, <span class=\\"bold\\">dataset</span>, <span class=\\"bold\\">datasheet</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">?</span></td><td data-title=\\"Example:\\">dat<span class=\\"highlight\\">?</span></td><td data-title=\\"Result:\\">Will find results containing words like <span class=\\"bold\\">data</span>, <span class=\\"bold\\">date</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">-</span></td><td data-title=\\"Example:\\">asia <span class=\\"highlight\\">-</span>china</td><td data-title=\\"Result:\\">Will find results containing the word  <span class=\\"bold\\">asia</span> that not mention the word <span class=\\"bold\\">china</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">NOT</span></td><td data-title=\\"Example:\\">asia <span class=\\"highlight\\">NOT</span>china</td><td data-title=\\"Result:\\">Will find results containing the word  <span class=\\"bold\\">asia</span> that not mention the word <span class=\\"bold\\">china</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">AND</span></td><td data-title=\\"Example:\\">paris <span class=\\"highlight\\">AND</span> london</td><td data-title=\\"Result:\\">Will find results that mention <span class=\\"bold\\">paris</span> and <span class=\\"bold\\">london</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">OR</span></td><td data-title=\\"Example:\\">paris <span class=\\"highlight\\">OR</span> london</td><td data-title=\\"Result:\\">Will find results containing the word  <span class=\\"bold\\">paris</span> or the word <span class=\\"bold\\">london</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">()</span></td><td data-title=\\"Example:\\"><span class=\\"highlight\\">(</span>europe AND paris<span class=\\"highlight\\">)</span> OR <span class=\\"highlight\\">(</span>europe AND london<span class=\\"highlight\\">)</span></td><td data-title=\\"Result:\\">Will find results that mention   <span class=\\"bold\\">europe</span> and  <span class=\\"bold\\">paris</span> or results that mention <span class=\\"bold\\">europe</span> and  <span class=\\"bold\\">london</span></td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">~</span></td><td data-title=\\"Example:\\">\\"heathrow airport\\"<span class=\\"highlight\\">~</span>3</td><td data-title=\\"Result:\\">Will find results where the words <span class=\\"bold\\">airport</span> and <span class=\\"bold\\">heathrow</span> appear within 3 words of each other.</td></tr><tr><td data-title=\\"Operator:\\"><span class=\\"highlight\\">sen:</span></td><td data-title=\\"Example:\\"><span class=\\"highlight\\">sen:</span>\\"Fashion & Style\\"</td><td data-title=\\"Result:\\">Will find results containing the phrase <span class=\\"bold\\">Fashion & Style</span></td></tr></tbody></table><div class=\\"definition-help-table-header-responsive\\">Example:</div><table><thead><tr><th>Example</th><th>Result</th></tr></thead><tbody><tr><td data-title=\\"Example:\\"><span class=\\"highlight\\">&quot;</span>fashion brands<span class=\\"highlight\\">&quot;</span> england</td><td data-title=\\"Result:\\">Will find results containing the phrase <span class=\\"bold\\">fashion brands</span> and the word <span class=\\"bold\\">england</span></td></tr><tr><td data-title=\\"Example:\\">europe <span class=\\"highlight\\">-</span>london <span class=\\"highlight\\">&quot;</span>airport gatwick<span class=\\"highlight\\">&quot;~</span>3</td><td data-title=\\"Result:\\">Will find results containing the word <span class=\\"bold\\">europe</span>, not containing the word <span class=\\"bold\\">london</span>, and where the words <span class=\\"bold\\">airport</span> and <span class=\\"bold\\">gatwick</span> appear within 3 words of each other.</td></tr><tr><td data-title=\\"Example:\\"><span class=\\"highlight\\">(&quot;</span>fashion show<span class=\\"highlight\\">&quot; AND (&quot;</span>new york<span class=\\"highlight\\">&quot; OR</span> paris<span class=\\"highlight\\">) AND (NOT</span> classic<span class=\\"highlight\\">))</span></td><td data-title=\\"Result:\\">Will find results containing the phrase <span class=\\"bold\\">fashion show</span> and the phrases  <span class=\\"bold\\">new york</span> or <span class=\\"bold\\">paris</span> and not containing the word <span class=\\"bold\\">classic</span></td></tr><tr><td data-title=\\"Example:\\"><span class=\\"highlight\\">(sen:&quot;</span>Fashion & Style<span class=\\"highlight\\">&quot; AND (sen:</span>@beauty<span class=\\"highlight\\"> OR</span> Beauty<span class=\\"highlight\\">)</span></td><td data-title=\\"Result:\\">Will find results containing the phrase <span class=\\"bold\\">Fashion & Style</span> and the words  <span class=\\"bold\\">@beauty</span> or <span class=\\"bold\\">beauty</span></td></tr></tbody></table></div></div></div></section></div></span>"`;

exports[`Definition social Component snapshot with spanish locales 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover\\" tabindex=\\"-1\\"><div aria-labelledby=\\"focusFeedDefinitionHelpDialogContainer\\" id=\\"focusFeedDefinitionHelpDialogContainer\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content\\"><span aria-hidden=\\"true\\"></span><div id=\\"focusFeedDefinitionHelpDialogContent\\"><div class=\\"definition-help-header\\"><button id=\\"focusFeedDefinitionHelpDialogCloseBtn\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon icon-cancel md-text--inherit\\"></i></button></div><div class=\\"definition-help-content\\"><div class=\\"definition-help-content-title\\"><i class=\\"md-icon material-icons\\">help_outline</i><div class=\\"definition-help-content-title-texts\\"><div class=\\"big-text\\">Feed definition help tips</div><div class=\\"small-text\\">Use these guidelines to improve your feed results</div></div></div><div class=\\"definition-help-content-tables\\"><div class=\\"definition-help-table-header-responsive\\">Operador:</div><table><thead><tr><th>Operador</th><th>Ejemplo</th><th>Resultado</th></tr></thead><tbody><tr><td data-title=\\"Operador:\\">&nbsp;</td><td data-title=\\"Ejemplo:\\">launchmetrics</td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezca la palabra <span class=\\"bold\\">launchmetrics</span></td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">\\"</span></td><td data-title=\\"Ejemplo:\\"><span class=\\"highlight\\">\\"</span>fashion brands<span class=\\"highlight\\">\\"</span></td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezcan juntas las palabras <span class=\\"bold\\">fashion brands</span></td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">*</span></td><td data-title=\\"Ejemplo:\\">camin<span class=\\"highlight\\">*</span></td><td data-title=\\"Resultado:\\">Encontrará resultados con palabras como <span class=\\"bold\\">camino</span>, <span class=\\"bold\\">camina</span>, <span class=\\"bold\\">caminos</span>, <span class=\\"bold\\">caminante</span></td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">?</span></td><td data-title=\\"Ejemplo:\\">camin<span class=\\"highlight\\">?</span></td><td data-title=\\"Resultado:\\">Encontrará resultados con palabras como <span class=\\"bold\\">camino</span>, <span class=\\"bold\\">camina</span></td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">-</span></td><td data-title=\\"Ejemplo:\\">europa <span class=\\"highlight\\">-</span>londres</td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezca la palabra  <span class=\\"bold\\">europa</span> y  no aparezca la palabra <span class=\\"bold\\">londres</span></td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">NOT</span></td><td data-title=\\"Ejemplo:\\">valencia <span class=\\"highlight\\">NOT</span> sevilla</td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezca  la palabra <span class=\\"bold\\">valencia</span> y no aparezca la palabra <span class=\\"bold\\">sevilla</span></td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">AND</span></td><td data-title=\\"Ejemplo:\\">paris <span class=\\"highlight\\">AND</span> london</td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezcan las palabras <span class=\\"bold\\">paris</span> y <span class=\\"bold\\">london</span></td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">OR</span></td><td data-title=\\"Ejemplo:\\">valencia <span class=\\"highlight\\">OR</span> sevilla</td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezca la palabra <span class=\\"bold\\">valencia</span> o la palabra <span class=\\"bold\\">sevilla</span></td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">()</span></td><td data-title=\\"Ejemplo:\\"><span class=\\"highlight\\">(</span>europa AND paris<span class=\\"highlight\\">)</span> OR <span class=\\"highlight\\">(</span>europa AND londres<span class=\\"highlight\\">)</span></td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezca <span class=\\"bold\\">europa</span> y <span class=\\"bold\\">paris</span> o resultados donde aparezca <span class=\\"bold\\">europa</span> y <span class=\\"bold\\">londres</span></td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">~</span></td><td data-title=\\"Ejemplo:\\">\\"aeropuerto barajas\\"<span class=\\"highlight\\">~</span>3</td><td data-title=\\"Resultado:\\">Encontrará resultados donde las palabras <span class=\\"bold\\">barajas</span> y <span class=\\"bold\\">aeropuerto</span> esten separadas un máximo de 3 palabras.</td></tr><tr><td data-title=\\"Operador:\\"><span class=\\"highlight\\">sen:</span></td><td data-title=\\"Ejemplo:\\"><span class=\\"highlight\\">sen:</span>\\"Fashion & Style\\"</td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezca la frase <span class=\\"bold\\">Fashion & Style</span></td></tr></tbody></table><div class=\\"definition-help-table-header-responsive\\">Ejemplo:</div><table><thead><tr><th>Ejemplo</th><th>Resultado</th></tr></thead><tbody><tr><td data-title=\\"Ejemplo:\\"><span class=\\"highlight\\">&quot;</span>fashion brands<span class=\\"highlight\\">&quot;</span> spain</td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezca la frase <span class=\\"bold\\">fashion brands</span> y la palabra <span class=\\"bold\\">spain</span></td></tr><tr><td data-title=\\"Ejemplo:\\">europe <span class=\\"highlight\\">-</span>london <span class=\\"highlight\\">&quot;</span>airport gatwick<span class=\\"highlight\\">&quot;~</span>3</td><td data-title=\\"Resultado:\\">Encontrará resultados donde apareza la palabra <span class=\\"bold\\">europa</span>, donde no apareza la palabra <span class=\\"bold\\">londres</span> y donde las palabras <span class=\\"bold\\">aeropuerto</span> y <span class=\\"bold\\">gatwick</span> esten separadas un máximo de 3 palabras.</td></tr><tr><td data-title=\\"Ejemplo:\\"><span class=\\"highlight\\">(&quot;</span>fashion show<span class=\\"highlight\\">&quot; AND (&quot;</span>new york<span class=\\"highlight\\">&quot; OR</span> paris<span class=\\"highlight\\">) AND (NOT</span> classic<span class=\\"highlight\\">))</span></td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezca la frase <span class=\\"bold\\">fashion show</span> junto con <span class=\\"bold\\">new york</span> o la frase <span class=\\"bold\\">paris</span> y no contenga la palabra <span class=\\"bold\\">classic</span></td></tr><tr><td data-title=\\"Ejemplo:\\"><span class=\\"highlight\\">(sen:&quot;</span>Fashion & Style<span class=\\"highlight\\">&quot; AND (sen:</span>@beauty<span class=\\"highlight\\"> OR</span> Beauty<span class=\\"highlight\\">)</span></td><td data-title=\\"Resultado:\\">Encontrará resultados donde aparezca la frase <span class=\\"bold\\">Fashion & Style</span> junto con <span class=\\"bold\\">@beauty</span> o <span class=\\"bold\\">beauty</span></td></tr></tbody></table></div></div></div></section></div></span>"`;
